OSDOCS-3969 Alibaba VPC installation - new assembly (#44593)
openshift/openshift-docs: modules/installation-custom-alibaba-vpc.adoc

// Module included in the following assemblies:
//
// * installing/installing_alibaba/installing-alibaba-vpc.adoc

:_content-type: CONCEPT
[id="installation-custom-alibaba-vpc_{context}"]
= About using a custom VPC

> In {product-title} {product-version}, you can deploy a cluster into existing subnets in an existing Virtual Private Cloud (VPC) in Alibaba Cloud Platform. By deploying {product-title} into an existing Alibaba VPC, you might be able to avoid limit constraints in new accounts or more easily abide by the operational constraints that your company's guidelines set. If you cannot obtain the infrastructure creation permissions that are required to create the VPC yourself, use this installation option. You must configure networking using vSwitches.

Reviewer: ```suggestion
In {product-title} {product-version}, you can deploy a cluster into existing subnets in an existing Virtual Private Cloud (VPC) in the Alibaba Cloud Platform. By deploying {product-title} into an existing Alibaba VPC, you might be able to avoid limit constraints in new accounts or more easily abide by the operational constraints that your company's guidelines set. If you cannot obtain the infrastructure creation permissions that are required to create the VPC yourself, use this installation option. You must configure networking using vSwitches.
```
```suggestion
In {product-title} {product-version}, you can deploy a cluster into existing subnets in an existing Virtual Private Cloud (VPC) in Alibaba Cloud Platform. By deploying {product-title} into an existing Alibaba VPC, you might be able to avoid limit constraints in new accounts or more easily abide by the operational constraints that your company's guidelines set. If you cannot obtain the infrastructure creation permissions that are required to create the VPC yourself, use this installation option. You must configure networking using vSwitches.
```
```suggestion
In {product-title} {product-version}, you can deploy a cluster into existing subnets in an existing Virtual Private Cloud (VPC) in the Alibaba Cloud Platform. By deploying {product-title} into an existing Alibaba VPC, you might be able to avoid limit constraints in new accounts or more easily abide by the operational constraints that adhere to your company's guidelines. If you cannot obtain the infrastructure creation permissions that are required to create the VPC yourself, use this installation option. You must configure networking using vSwitches.
```

Reviewer: oops , please look at the last suggestion :)

Author: OK! Got it, all good, thank you!

[id="installation-custom-alibaba-vpc-requirements_{context}"]
== Requirements for using your VPC

The union of the VPC CIDR block and the machine network CIDR must be non-empty. The vSwitches must be within the machine network.

The installation program does not create the following components:

* VPC
* vSwitches
* Route table
* NAT gateway

include::snippets/custom-dns-server.adoc[]

[id="installation-custom-alibaba-vpc-validation_{context}"]
== VPC validation

To ensure that the vSwitches you provide are suitable, the installation program confirms the following data:

* All the vSwitches that you specify must exist.

> * You provide one or more vSwitches for control plane machines and compute machines.

Reviewer: ```suggestion
* You have provided one or more vSwitches for control plane machines and compute machines.
```

Author: Done, thank you!

* The vSwitch's CIDRs belong to the machine CIDR that you specified.

[id="installation-about-custom-alibaba-permissions_{context}"]
== Division of permissions

> Some individuals can create different resources in your cloud than others. For example, you might be able to create application-specific items, like instances, buckets, and load balancers, but not networking-related components such as VPCs or vSwitches.

Reviewer: ```suggestion
Some individuals can create different resources in your cloud than others. For example, you might be able to create application-specific items, like instances, buckets, and load balancers, but not networking-related components, such as VPCs or vSwitches.
```

Author: Done, thank you!

[id="installation-custom-alibaba-vpc-isolation_{context}"]
== Isolation between clusters

> If you deploy {product-title} to an existing network, the isolation of cluster services is reduced in the following ways:

Reviewer: I am not sure , is t_o _to an_ existing network or into an existing network_.

Author: I'll accept your suggested change. It seems like it might aid localization.

* You can install multiple {product-title} clusters in the same VPC.

> * ICMP ingress is allowed to the entire network.

Reviewer: into vs to the network

Author: I'm going to leave that one as is, because I believe it is "access to" and not "access into". In this case ICMP ingress is an access permission to a network, not access permission into a network. It's different than the usage above because this is not deployment to, it is a recurring access to.
